specs/obmd-integration.md

```diff
@@ -0,0 +1,140 @@
+# Problem
+
+As discussed in [Issue #417: Serial console support is horribly
+fragile][417], the current implementation of serial console support is
+not very robust, has known resource leaks, and is generally pretty
+janky.
+
+We are developing a separate daemon (called obmd) to manage
+the OBMs for nodes. An early version is already available at
+<https://github.com/zenhack/obmd>. Rationale is discussed in more detail
+in the "Alternative Solutions" section below.
+
+This document proposes changes to HIL to integrate with obmd.
+
+# Solution
+
+## Summary
+
+* OBMs will no longer be drivertized, further discussion below.
+* `node_register` will be modified to accept obmd connection info:
+  * an admin token
+  * a base URL for the node, e.g.
+    `https://obmd.example.com/node/mynode`.
+* The api calls `start_console` and `stop_console` will be removed.
+* Two new api calls will be added:
+  * `enable_obm`, which will fetch a token for the node's OBM
+     from obmd, storing it in the database. If there is already
+     a valid token, it will be silently replaced.
```

Comment:
Why are you replacing the token during `enable_obm` rather than telling the user that the obm is already enabled?  Would there be a case when a token could be invalidated and need replacing?